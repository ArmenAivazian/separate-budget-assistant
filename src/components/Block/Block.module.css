.block-grid {
    display: grid;
    gap: 10px;
    grid-template-columns: auto 1fr;
}