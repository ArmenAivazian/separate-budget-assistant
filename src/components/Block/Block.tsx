import { ConfigValue } from "../../types/ConfigValue";
import { Title } from "../Title";
import style from "./Block.module.css";
import { Field } from "./components/Field";

export function Block({ title, list }: { title: string; list: ConfigValue[] }) {
  return (
    <>
      <Title text={title} />
      <div className={style["block-grid"]}>
        {list.map((configs) => (
          <Field key={configs.text} {...configs} />
        ))}
      </div>
    </>
  );
}
