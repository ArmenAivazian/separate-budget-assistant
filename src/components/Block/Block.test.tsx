import renderer from "react-test-renderer";

import { Block } from "./Block";

describe("Block", () => {
  describe("snapshots", () => {
    it("default", () => {
      const component = renderer.create(
        <Block
          title="title"
          list={[
            { value: "value", text: "text", setValue: () => {} },
            { value: "value2", text: "text2", setValue: () => {} },
          ]}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
