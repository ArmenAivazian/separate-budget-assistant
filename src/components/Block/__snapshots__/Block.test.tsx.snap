// Vitest Snapshot v1

exports[`Block > snapshots > default 1`] = `
[
  <h1
    className="_title_d28f04"
  >
    title
  </h1>,
  <div
    className="_block-grid_17898c"
  >
    <p
      className="_field-name_10d9f7"
    >
      text
       :
    </p>
    <input
      onChange={[Function]}
      type="number"
      value="value"
    />
    <p
      className="_field-name_10d9f7"
    >
      text2
       :
    </p>
    <input
      onChange={[Function]}
      type="number"
      value="value2"
    />
  </div>,
]
`;
